rules-reviewed/eap7/eap6/jax-ws.windup.xml

```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0"?>
+<ruleset xmlns="http://windup.jboss.org/schema/jboss-ruleset" id="jax-ws"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+
+    <metadata>
+        <description>
+            This ruleset provides analysis of applications that use JAX-WS and may require
+            individual attention when migrating to JBoss EAP 7+.
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-javaee,3.0.0.Final" />
+            <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final" />
+        </dependencies>
+        <sourceTechnology id="eap" versionRange="[6,7)" />
+        <targetTechnology id="eap" versionRange="[7,)" />
+    </metadata>
+
+    <rules>
+        <rule id="jax-ws-00000">
+            <when>
+                <and>
+                    <javaclass references="javax.xml.ws.Service" as="default">
+                        <location>INHERITANCE</location>
+                    </javaclass>
+                    <not>
+                        <javaclass from="default" references="{*}(java.net.URL, javax.xml.namespace.QName, javax.xml.ws.WebServiceFeature)"/>
```

Comment:
Correct identification of new added constructor in JAX-WS 2.2 style is
```xml
<javaclass references="{*}(java.net.URL, javax.xml.namespace.QName, javax.xml.ws.WebServiceFeature{*})">
     <location>METHOD</location>
</javaclass>
```
But `<not>` is not working as expected

Reply:
The problem with not in the "when" is that the semantics are naturally a bit strange. When the first "javaclass" returns results, those will be fed through the second "javaclass" condition. 

If the second javaclass condition matches anything and has results then it will return true. The not will turn this into a false and the rule's "perform" will not be performed.

If the second javaclass condition does not match anything and has zero results, then it will return false. The not will flip this to a true, but I'm pretty sure that there will now be zero results returned. The perform may fire, but it likely will not have anything to iterate over.

I think you might be able to get all constructors and just use an iteration filter?